pkg/proto/agentbaker/v1/config.proto

```diff
@@ -0,0 +1,229 @@
+syntax = "proto3";
+
+package agentbaker.v1;
+
+// illustrative division only
+message Configuration {
+  // System Configuration
+  
+  string cluster_certificate_authority = 1;
+
+  // only required until Secure TLS bootstrapping in place
+  // would use kubelet identity after that.
+  string tls_bootstrap_token = 2;
+
+  // cluster/user config
+  string kubernetes_version = 3; // Q: can this be auto-detected? Or is this part of specifying the desired node version?
+    // removed kubernetesHyperkubeSpec
+  string kube_binary_url = 4;      // maybe we can combine this and custom_kube_binary_url later once we figure out the logic of how they are used.
+  string custom_kube_binary_url = 5;
+  string kubeproxy_url = 6;
+  ApiServerConfig apiserver_config = 7;
+  string subscription_id = 8;
+  string resource_group = 9;
+  string location = 10;
+  string vm_type = 11;  
+  
+  string primary_availability_set = 12;
+  string primary_scale_set = 13;
+  
+  CloudProviderConfig cloud_provider_config = 14;
+  IdentityConfiguration identity_configuration = 15;
+
+  RuncConfig runc_config = 16;
+
+  // ... and a bunch of user-specified config
+  string repo_depot_endpoint = 17;
+  string linux_admin_username = 18;
+    // removed mobyVersion
+  string tenant_id = 19;
+  bool use_instance_metadata = 20;
+  LoadBalancerConfig load_balancer_config = 21;
+    // removed containerRuntime
+    // removed cliTool because docker is deprecated. Only "ctr" will be supported.
+  ContainerdConfig containerd_config = 22;
+  bool is_vhd = 23;
+  GpuConfig gpu_config = 24;
+  bool is_sgx_node = 25;
+  TeleportConfig teleport_config = 26;
+  // removed NEEDS_CONTAINERD because it's always needed now
+
+  HttpProxyConfig http_proxy_configuration = 27;
+  repeated string trusted_certificate_authorities = 28;
```

Comment:
What is this used/needed for? Is it for custom CA trust feature? If yes, we also have `repeated string custom_ca_certs = 3;`, do we need that separate field? Or is it for something else?

Reply:
> What is this used/needed for? Is it for custom CA trust feature? If yes, we also have `repeated string custom_ca_certs = 3;`, do we need that separate field? Or is it for something else?

Thanks for catching this. Yes this is duplicate.